src/index.js

```diff
@@ -106,6 +106,14 @@ const UserAuthWrapper = (args) => {
 
     return hoistStatics(UserAuthWrapper, DecoratedComponent)
   }
+
+  wrapComponent.onEnter = (store, nextState, replace) => {
+    const replaceWithState = ({ pathname, query }) => replace({ pathname, state: query })
```

Comment:
From what I read on the react-router doc, to use query string instead of state would be done by replacing previous line with:

``` javascript
const replaceWithState = ({ pathname, query }) => replace({ pathname, query })
```